core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1068,13 +1068,14 @@ class KafkaApis(val requestChannel: RequestChannel,
             response
           } catch {
             // NOTE: These exceptions are special cases since these error messages are typically transient or the client
-            // would have received a clear exception and there is no value in logging the entire stack trace for the same
+            // would have received a clear exception.
             case e @ (_ : UnknownTopicOrPartitionException |
-                      _ : NotLeaderForPartitionException |
+                      _ : NotLeaderOrFollowerException |
                       _ : UnknownLeaderEpochException |
                       _ : FencedLeaderEpochException |
                       _ : KafkaStorageException |
                       _ : UnsupportedForMessageFormatException) =>
+              // print the stack trace for debugging
               e.printStackTrace()
```

Comment:
This doesn't make much sense, we should not be printing the stacktrace like this.

Reply:
I agree. But I'd like to confirm with @mimaison since he added it in #8295. Thanks.